tests/unit/geo.tcl

```diff
@@ -456,4 +589,92 @@ start_server {tags {"geo"}} {
         }
         set test_result
     } {OK}
+    }
+
+    test {GEOSEARCH box edges fuzzy test} {
+        if {$::accurate} { set attempt 300 } else { set attempt 30 }
+        while {[incr attempt -1]} {
+            unset -nocomplain debuginfo
+            set srand_seed [clock milliseconds]
+            lappend debuginfo "srand_seed is $srand_seed"
+            expr {srand($srand_seed)} ; # If you need a reproducible run
+            r del mypoints
+
+            geo_random_point search_lon search_lat
+            set width_m [expr {[randomInt 10000]+10}]
+            set height_m [expr {[randomInt 10000]+10}]
+            set lat_delta [geo_raddeg [expr {$height_m/2/6372797.560856}]]
+            set long_delta_top [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat+$lat_delta}]])}]]
+            set long_delta_middle [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad $search_lat])}]]
+            set long_delta_bottom [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat-$lat_delta}]])}]]
+
+            # Total of 8 points are generated, which are located at each vertex and the center of each side
+            set points(0) [expr {$search_lon-$long_delta_top}]
+            set points(1) [expr {$search_lat+$lat_delta}]
+            set points(2) [expr {$search_lon+$long_delta_top}]
+            set points(3) [expr {$search_lat+$lat_delta}]
+            set points(4) [expr {$search_lon+$long_delta_bottom}]
+            set points(5) [expr {$search_lat-$lat_delta}]
+            set points(6) [expr {$search_lon-$long_delta_bottom}]
+            set points(7) [expr {$search_lat-$lat_delta}]
+            set points(8) $search_lon
+            set points(9) [expr {$search_lat+$lat_delta}]
+            set points(10) $search_lon
+            set points(11) [expr {$search_lat-$lat_delta}]
+            set points(12) [expr {$search_lon+$long_delta_middle}]
+            set points(13) $search_lat
+            set points(14) [expr {$search_lon-$long_delta_middle}]
+            set points(15) $search_lat
+
+            lappend debuginfo "Search area: geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_m $height_m m"
+            set tcl_result {}
+            for {set i 0} {$i < 16} {incr i 2} {
+                set x $points($i)
+                set y $points([expr {$i+1}])
+                r geoadd mypoints $x $y place:$i
+                lappend tcl_result "place:$i"
+                lappend debuginfo "geoadd mypoints $x $y place:$i"
+            }
```

Comment:
i think that instead of this long list of numbers (some represent long and some represent lat), and then a loop that work on pairs), it would be nicer to create a proc that takes name lon lat adds a point.
this way, we can name them north, south, north-west, etc, and the code that skips points below can be clearer (not referring to "place:12" and "place:14"